### PR TITLE
Disable Metrics table auto split

### PR DESCRIPTION
**Pull request by CuriousVini**
Issue: https://issues.cask.co/browse/CDAP-12591
Build: https://builds.cask.co/browse/CDAP-RUT1385-11

Tested on fresh and upgraded cluster.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/hbase/AbstractHBaseDataSetAdmin.java [hunk 1]
```diff
@@ -234,7 +234,7 @@ protected void addCoprocessor(HTableDescriptorBuilder tableDescriptor, Class<? e
    *
    * @return true if the table descriptor is modified, that is, whether update is needed.
    */
-  protected abstract boolean needsUpdate(HTableDescriptor tableDescriptor);
+  protected abstract boolean needsUpdate(HTableDescriptor tableDescriptor, HTableDescriptorBuilder descriptorBuilder);
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/hbase/AbstractHBaseDataSetAdmin.java]: Add `@param` to tell what are `tableDescriptor` and `descriptorBuilder` are for.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java [hunk 2]
```diff
@@ -184,6 +193,16 @@ protected boolean needsUpdate(HTableDescriptor tableDescriptor) {
       needUpgrade = true;
     }
 
+
+    String splitsPolicy = spec.getProperty(SPLIT_POLICY);
+    // override using provided split policy
+    if (!Strings.isNullOrEmpty(splitsPolicy) &&
+      (tableDescriptor.getRegionSplitPolicyClassName() == null ||
+        !tableDescriptor.getRegionSplitPolicyClassName().equalsIgnoreCase(splitsPolicy))) {
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java]: You can simplify the condition to

```
if (!Strings.isNullOrEmpty(splitsPolicy) && !splitsPolicy.equals(tableDescriptor.getRegionSplitPolicyClassName()))
```

Also, class name should be compared with `equals`, not `equalsIgnoreCase`.

**Comment by chtyim** [comment]
Just two simple comments.

**Comment by CuriousVini** [comment]
@chtyim Addressed comments. Please take another pass when you get a chance, thanks

**Comment by chtyim** [comment]
LGTM

**Comment by CuriousVini** [comment]
Build is green. Merging.

**Pull request merged**